kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/http/AbstractSimultaneousConnectionsTest.java

```diff
@@ -114,6 +114,10 @@ public void http1Connections() throws Exception {
           .build();
       for (int it = 0; it < MAX_HTTP_1_CONNECTIONS; it++) {
         asyncResponses.add(client.consumeBytes(request, (value, asyncBody) -> asyncBody.consume()));
+        // Throttle connection creation to keep cool with the server otherwise it might reset connections
```

Comment:
@manusa @vietj there's also #4736 - in that I didn't want to make any assumption about rate limiting the creation, so instead I added logic to keep retrying if there are failures.  

OkHttp must be doing something different internally as it didn't seem to trigger any ill-effects on the server-side - but jdk, jetty, and vertx seem to cause flooding that isn't handled at least for websockets.  The original exceptions for vertx in these tests though seemed to be stackoverflows - is there something more severe going on once vertx starts seeing reset connections?

I'm not opposed to slowing down the creation in these tests - it will help with the jdk async thread pool usage as well - just want to make sure there's not other considerations in play.

Reply:
That does not slow a lot the tests according to my tests, the tests are still very fast, the HTTP/1 test takes 250ms on my machine to pass.

Jetty might throttle internally the creation of connection.

Reply:
> That does not slow a lot the tests according to my tests, the tests are still very fast, the HTTP/1 test takes 250ms on my machine to pass.

I'm less concerned with the runtime of the test and how any issue with 500 from the server is handled.

Without these changes, here's what's in the log from a run of the tests https://gist.github.com/shawkins/4f174fb0c25d30f5524f42a9ea5d3cba - but I'm not sure if that's directly due to the test run, shutdown, etc.

Reply:
interesting it looks like a bug in Vertx that happens in that case, thanks I'll report it and we'll get a fix for that

Reply:
I think it happens because we have many connection on the same pool that cannot connect subsequently

Reply:
Checking #4736